sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -659,13 +660,19 @@ case class StructsToJson(
         (arr: Any) =>
           gen.write(arr.asInstanceOf[ArrayData])
           getAndReset()
+      case MapType(_: DataType, _: StructType, _: Boolean) =>
+        (map: Any) =>
+          val mapType = child.dataType.asInstanceOf[MapType]
+          gen.write(map.asInstanceOf[MapData], mapType)
+          getAndReset()
     }
   }
 
   override def dataType: DataType = StringType
 
   override def checkInputDataTypes(): TypeCheckResult = child.dataType match {
-    case _: StructType | ArrayType(_: StructType, _) =>
+    case _: StructType | ArrayType(_: StructType, _) |
+         MapType(_: DataType, _: StructType, _: Boolean) =>
```

Comment:
yeah, just like @viirya said. This expression is named `StructsToJson`.
Another reason is that it use `JacksonGenerator` which will need a `rowSchema`  when initializing to generate JSON string. Therefore, it must be restricted to `StructType` here.
or can we create another expression to support arbitrary map type?